File: week6/sentimental-readability/readability.py
# TODO
from cs50 import get_string


def main():

    text = get_string("Text: ")

    # Count the number of letters
    letters = num_letters(text)

    # Count the number of words
    words = num_words(text)

    # Count the number of sentences
    sentences = num_sentences(text)

    """ Stores the average number of letters per
    100 words in the text in a varirable """
    L = (letters * 100) / words

    """ Stores the average number of sentences per
    100 words in the text in a varirable """
    S = (sentences * 100) / words

    # Identifies grade level using the formula
    index = 0.0588 * L - 0.296 * S - 15.8

    # Round the result gotten by the formula
    index = round(index)

    # Print the grade level
    if index > 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")


# Count the number of letters in the text provided
def num_letters(text):
    i = 0
    for c in text:
        if c.isalpha() == True:
            i += 1
    return i


def num_words(text):
    # Count the number of words in the text provided
    i = 1
    for c in text:
        if c == ' ':
            i += 1
    return i


def num_sentences(text):
    # Count the number of sentences in the text provided
    i = 0
    for c in text:
        if c == '.' or c == '?' or c == '!':
            i += 1
    return i


main()

File: week2/readability/readability.c
#include <cs50.h>
#include <math.h>
#include <string.h>
#include <stdio.h>

// Function that counts the letters in a text
int count_letters(string text);

// Function that counts the words in a text
int count_words(string text);

// Function that counts the sentences in a text
int count_sentences(string text);

int main(void)
{
    // Prompt user for a string
    string text = get_string("Text: ");

    // Store the number of letters, words, and sentences in the text in three variables
    float letters = count_letters(text);
    float words = count_words(text);
    float sentences = count_sentences(text);

    // Stores the average number of letters per 100 words in the text in a varirable
    float L = (letters * 100) / words;

    // Stores the average number of sentences per 100 words in the text in a varirable
    float S = (sentences * 100) / words ;

    // Identifies the grade level by using the formula
    float f = 0.0588 * L - 0.296 * S - 15.8;

    // Rounds the result
    int index = round(f);

    // Prints the grade level
    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }

}





int count_letters(string text)
{
    int j = 0;
    for (int i = 0, length = strlen(text); i < length; i++)
    {
        if (text[i] >= 'a' && text[i] <= 'z')
        {
            j++;
        }
        if (text[i] >= 'A' && text[i] <= 'Z')
        {
            j++;
        }
    }

    return j;
}

int count_words(string text)
{
    int j = 0;
    for (int i = 0, length = strlen(text); i < length; i++)
    {

        if (text[i] == ' ')
        {
            j++;
        }
    }
    j++;
    return j;
}

int count_sentences(string text)
{
    int j = 0;
    for (int i = 0, length = strlen(text); i < length; i++)
    {

        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            j++;
        }
    }

    return j;
}
